src/Fields/broadcast.jl

```diff
@@ -279,6 +279,13 @@ Base.Broadcast.broadcasted(
     args...,
 ) where {T} = Base.Broadcast.broadcasted(fs, (x...) -> T(x...), args...)
 
+# GPU support for type wrappers, like Geometry.UVVector(1, 2)
+Base.Broadcast.broadcasted(
```

Comment:
CI is erroring because this introduces new method ambiguities (in the aqua tests). We can either fix the ambiguities or increase our limit. I’d prefer we not introduce new ones if it’s an easy fix.

Let’s see if narrowing the tapes for the second argument, fixes the issue, if not just revert and include the ambiguities, and increase the limit to what the tests says on the remote machine (assuming you can’t reproduce the aqua tests locally).

Reply:
You mean narrowing the type of the second argument? But we don't do that in the previous `broadcasted` method definition that dispatches on `AbstractFieldStyle`

Reply:
Anyway, if you meant type,  tried to specify the second argument type, as in
```
Base.Broadcast.broadcasted(
    fs::Base.Broadcast.DefaultArrayStyle{0},
    ::Type{Geometry.UVVector},
    args...,
) = Base.Broadcast.broadcasted(fs, (x...) -> Geometry.UVVector(x...), args...)
```
but the ambiguities are still 15, rather than 16 like in the failing CI case, but still breaking the aqua test (that tests  ≤ 10)

Reply:
Ok. Let's narrow the second argument types to AxisTensors.

Check if there's ambiguities with that. If not, try using the AbstractArrayStyle for the first argument, and see if there are ambiguities with that.

For more info on ambiguities: https://docs.julialang.org/en/v1/manual/methods/#man-ambiguities

Reply:
> Using ` AxisTensor` won't work either. The only way it is working (with the least amount of ambiguities is the way I pushed now):
> 
> ```
> Base.Broadcast.broadcasted(
>     fs::Base.Broadcast.DefaultArrayStyle{0},
>     ::Type{Geometry.UVVector},
>     args...,
> ) = Base.Broadcast.broadcasted(fs, (x...) -> Geometry.UVVector(x...), args...)
> ```

Can you show me what you tried? And what's the failure?

Reply:
1. 

The suggested version with `AxisTensor` in the second argument, still breaks the test with the same error as in the original version. That is:
```
Base.Broadcast.broadcasted(
    fs::Base.Broadcast.DefaultArrayStyle{0},
    ::Geometry.AxisTensor,
    args...,
) = Base.Broadcast.broadcasted(fs, (x...) -> Geometry.AxisTensor(x...), args...)
```
gives: 
```
$ julia --project=test test/Operators/hybrid/cuda.jl 
running test on ClimaComms.CUDADevice() device
Test Summary:                      | Pass  Total   Time
Finite difference GradientF2C CUDA |    1      1  39.3s
running test on ClimaComms.CUDADevice() device
2D SE, 1D FD Extruded Domain ∇ ODE Solve horizontal CUDA: Error During Test at /home/valeria/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:102
  Got exception outside of a @test
  GPU compilation of MethodInstance for ClimaCore.Operators.copyto_spectral_kernel!(::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, ::Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}) failed
  KernelError: passing and using non-bitstype argument
  
  Argument 3 to your kernel function is of type Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}, which is not isbits:
    .args is of type Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}} which is not isbits.
      .1 is of type ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing} which is not isbits.
        .args is of type Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}} which is not isbits.
          .1 is of type Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}} which is not isbits.
            .args is of type Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}} which is not isbits.
              .2 is of type Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}} which is not isbits.
                .f is of type Type{ClimaCore.Geometry.UVVector} which is not isbits.
  
  
  Stacktrace:
    [1] check_invocation(job::GPUCompiler.CompilerJob)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/validation.jl:92
    [2] macro expansion
      @ ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:101 [inlined]
    [3] macro expansion
      @ ~/.julia/packages/TimerOutputs/RsWnF/src/TimerOutput.jl:253 [inlined]
    [4] codegen(output::Symbol, job::GPUCompiler.CompilerJob; libraries::Bool, deferred_codegen::Bool, optimize::Bool, cleanup::Bool, strip::Bool, validate::Bool, only_entry::Bool, parent_job::Nothing, ctx::LLVM.Context)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:99
    [5] compile(target::Symbol, job::GPUCompiler.CompilerJob; libraries::Bool, deferred_codegen::Bool, optimize::Bool, cleanup::Bool, strip::Bool, validate::Bool, only_entry::Bool, ctx::LLVM.Context)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:90
    [6] compile(job::GPUCompiler.CompilerJob, ctx::LLVM.Context)
      @ CUDA ~/.julia/packages/CUDA/p5OVK/src/compiler/compilation.jl:105
    [7] #201
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/compilation.jl:100 [inlined]
    [8] JuliaContext(f::CUDA.var"#201#202"{GPUCompiler.CompilerJob{GPUCompiler.PTXCompilerTarget, CUDA.CUDACompilerParams}})
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:37
    [9] compile
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/compilation.jl:99 [inlined]
   [10] actual_compilation(cache::Dict{UInt64, Any}, key::UInt64, cfg::GPUCompiler.CompilerConfig{GPUCompiler.PTXCompilerTarget, CUDA.CUDACompilerParams}, ft::Type, tt::Type, compiler::typeof(CUDA.compile), linker::typeof(CUDA.link))
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/cache.jl:53
   [11] cached_compilation(cache::Dict{UInt64, Any}, cfg::GPUCompiler.CompilerConfig{GPUCompiler.PTXCompilerTarget, CUDA.CUDACompilerParams}, ft::Type, tt::Type, compiler::Function, linker::Function)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/cache.jl:37
   [12] macro expansion
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:310 [inlined]
   [13] macro expansion
      @ ./lock.jl:223 [inlined]
   [14] cufunction(f::typeof(ClimaCore.Operators.copyto_spectral_kernel!), tt::Type{Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}}}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
      @ CUDA ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:306
   [15] cufunction(f::typeof(ClimaCore.Operators.copyto_spectral_kernel!), tt::Type{Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}}})
      @ CUDA ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:303
   [16] macro expansion
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:104 [inlined]
   [17] copyto!(out::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, sbc::Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}, Nothing}}})
      @ ClimaCore.Operators ~/CliMA/ClimaCore.jl/src/Operators/spectralelement.jl:250
   [18] materialize!
      @ ./broadcast.jl:871 [inlined]
   [19] materialize!
      @ ./broadcast.jl:868 [inlined]
   [20] (::var"#rhs!#1")(dudt::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, u::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, #unused#::Nothing, t::Float64)
      @ Main ~/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:114
   [21] macro expansion
      @ ~/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:126 [inlined]
   [22] macro expansion
      @ ~/julia-1.8.5/share/julia/stdlib/v1.8/Test/src/Test.jl:1363 [inlined]
   [23] top-level scope
      @ ~/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:111
   [24] include(mod::Module, _path::String)
      @ Base ./Base.jl:419
   [25] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:303
   [26] _start()
      @ Base ./client.jl:522
Test Summary:                                            | Error  Total   Time
2D SE, 1D FD Extruded Domain ∇ ODE Solve horizontal CUDA |     1      1  11.5s
ERROR: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/valeria/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:102
```

2. 

The other suggestion of using `::Geometry.UVVector` also produces the same error:
```
Base.Broadcast.broadcasted(
    fs::Base.Broadcast.DefaultArrayStyle{0},
    ::Geometry.UVVector,
    args...,
) = Base.Broadcast.broadcasted(fs, (x...) -> Geometry.UVVector(x...), args...)
```
gives:
```
julia --project=test test/Operators/hybrid/cuda.jl 
running test on ClimaComms.CUDADevice() device
Test Summary:                      | Pass  Total   Time
Finite difference GradientF2C CUDA |    1      1  36.1s
running test on ClimaComms.CUDADevice() device
2D SE, 1D FD Extruded Domain ∇ ODE Solve horizontal CUDA: Error During Test at /home/valeria/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:102
  Got exception outside of a @test
  GPU compilation of MethodInstance for ClimaCore.Operators.copyto_spectral_kernel!(::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, ::Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}) failed
  KernelError: passing and using non-bitstype argument
  
  Argument 3 to your kernel function is of type Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}, which is not isbits:
    .args is of type Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}} which is not isbits.
      .1 is of type ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing} which is not isbits.
        .args is of type Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}} which is not isbits.
          .1 is of type Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}} which is not isbits.
            .args is of type Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}} which is not isbits.
              .2 is of type Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}} which is not isbits.
                .f is of type Type{ClimaCore.Geometry.UVVector} which is not isbits.
  
  
  Stacktrace:
    [1] check_invocation(job::GPUCompiler.CompilerJob)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/validation.jl:92
    [2] macro expansion
      @ ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:101 [inlined]
    [3] macro expansion
      @ ~/.julia/packages/TimerOutputs/RsWnF/src/TimerOutput.jl:253 [inlined]
    [4] codegen(output::Symbol, job::GPUCompiler.CompilerJob; libraries::Bool, deferred_codegen::Bool, optimize::Bool, cleanup::Bool, strip::Bool, validate::Bool, only_entry::Bool, parent_job::Nothing, ctx::LLVM.Context)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:99
    [5] compile(target::Symbol, job::GPUCompiler.CompilerJob; libraries::Bool, deferred_codegen::Bool, optimize::Bool, cleanup::Bool, strip::Bool, validate::Bool, only_entry::Bool, ctx::LLVM.Context)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:90
    [6] compile(job::GPUCompiler.CompilerJob, ctx::LLVM.Context)
      @ CUDA ~/.julia/packages/CUDA/p5OVK/src/compiler/compilation.jl:105
    [7] #201
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/compilation.jl:100 [inlined]
    [8] JuliaContext(f::CUDA.var"#201#202"{GPUCompiler.CompilerJob{GPUCompiler.PTXCompilerTarget, CUDA.CUDACompilerParams}})
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/driver.jl:37
    [9] compile
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/compilation.jl:99 [inlined]
   [10] actual_compilation(cache::Dict{UInt64, Any}, key::UInt64, cfg::GPUCompiler.CompilerConfig{GPUCompiler.PTXCompilerTarget, CUDA.CUDACompilerParams}, ft::Type, tt::Type, compiler::typeof(CUDA.compile), linker::typeof(CUDA.link))
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/cache.jl:53
   [11] cached_compilation(cache::Dict{UInt64, Any}, cfg::GPUCompiler.CompilerConfig{GPUCompiler.PTXCompilerTarget, CUDA.CUDACompilerParams}, ft::Type, tt::Type, compiler::Function, linker::Function)
      @ GPUCompiler ~/.julia/packages/GPUCompiler/BxfIW/src/cache.jl:37
   [12] macro expansion
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:310 [inlined]
   [13] macro expansion
      @ ./lock.jl:223 [inlined]
   [14] cufunction(f::typeof(ClimaCore.Operators.copyto_spectral_kernel!), tt::Type{Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}}}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
      @ CUDA ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:306
   [15] cufunction(f::typeof(ClimaCore.Operators.copyto_spectral_kernel!), tt::Type{Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{Nothing, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuDeviceArray{Float64, 4, 1}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuDeviceArray{Float64, 3, 1}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.DeviceIntervalTopology{NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuDeviceArray{Float64, 5, 1}}}, Nothing}}}}})
      @ CUDA ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:303
   [16] macro expansion
      @ ~/.julia/packages/CUDA/p5OVK/src/compiler/execution.jl:104 [inlined]
   [17] copyto!(out::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, sbc::Base.Broadcast.Broadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}, typeof(ClimaCore.RecursiveApply.rsub), Tuple{ClimaCore.Operators.SpectralBroadcasted{ClimaCore.Operators.CUDASpectralStyle, ClimaCore.Operators.Divergence{(1, 2)}, Tuple{Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VIJFHStyle{8, CuArray{Float64, N, CUDA.Mem.DeviceBuffer} where N}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}, typeof(ClimaCore.RecursiveApply.rmul), Tuple{ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, Type{ClimaCore.Geometry.UVVector}, Tuple{Float64, Float64}}}}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}, Nothing}}})
      @ ClimaCore.Operators ~/CliMA/ClimaCore.jl/src/Operators/spectralelement.jl:250
   [18] materialize!
      @ ./broadcast.jl:871 [inlined]
   [19] materialize!
      @ ./broadcast.jl:868 [inlined]
   [20] (::var"#rhs!#1")(dudt::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, u::ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.RectilinearMesh{ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.XPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.XPoint{Float64}, Int64}}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.YPoint{Float64}, Nothing}, LinRange{ClimaCore.Geometry.YPoint{Float64}, Int64}}}, CartesianIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}, CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{8}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.XYPoint{Float64}, Float64, SMatrix{2, 2, Float64, 4}}, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float64, 8, CuArray{Float64, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 8, CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, LinRange{ClimaCore.Geometry.ZPoint{Float64}, Int64}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.XYZPoint{Float64}, Float64, SMatrix{3, 3, Float64, 9}}, 8, CuArray{Float64, 5, CUDA.Mem.DeviceBuffer}}}}, #unused#::Nothing, t::Float64)
      @ Main ~/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:114
   [21] macro expansion
      @ ~/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:126 [inlined]
   [22] macro expansion
      @ ~/julia-1.8.5/share/julia/stdlib/v1.8/Test/src/Test.jl:1363 [inlined]
   [23] top-level scope
      @ ~/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:111
   [24] include(mod::Module, _path::String)
      @ Base ./Base.jl:419
   [25] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:303
   [26] _start()
      @ Base ./client.jl:522
Test Summary:                                            | Error  Total   Time
2D SE, 1D FD Extruded Domain ∇ ODE Solve horizontal CUDA |     1      1  11.1s
ERROR: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/valeria/CliMA/ClimaCore.jl/test/Operators/hybrid/cuda.jl:102
```

3. 

The only versions that do work are the current one:
```
Base.Broadcast.broadcasted(
    fs::Base.Broadcast.DefaultArrayStyle{0},
    ::Type{Geometry.UVVector},
    args...,
) = Base.Broadcast.broadcasted(fs, (x...) -> Geometry.UVVector(x...), args...)
```
that passes the cuda test but produces 15 ambiguities in the aqua test. 

4. 
And the previous, generic version
```
Base.Broadcast.broadcasted(
    fs::Base.Broadcast.DefaultArrayStyle{0},
    ::Type{T},
    args...,
) where {T} = Base.Broadcast.broadcasted(fs, (x...) -> T(x...), args...)
```
that passes the cuda test but produced 16 ambiguities in the aqua test. 

5. 

 I also tried the generic version with the `DefaultArrayStyle` on the first argument, generic one on the second:
 ```
 Base.Broadcast.broadcasted(
    fs::Base.Broadcast.DefaultArrayStyle{0},
    ::Type{T},
    args...,
) where {T} = Base.Broadcast.broadcasted(fs, (x...) -> T(x...), args...)
 ```
 but produces 15 ambiguities (as the current specialized version). 
 
 6. 
 I also tried the version with `AbstractArrayStyle` on the first argument, generic one on the second:
 ```
 Base.Broadcast.broadcasted(
    fs::Base.Broadcast.AbstractArrayStyle,
    ::Type{T},
    args...,
) where {T} = Base.Broadcast.broadcasted(fs, (x...) -> T(x...), args...)
 ```
 and produces 15 ambiguities.
 
 7. 
 Finally, the version with `AbstractArrayStyle` on the first argument and more specialized on the second, such as:
  ```
Base.Broadcast.broadcasted(
    fs::Base.Broadcast.AbstractArrayStyle,
    ::Type{Geometry.UVVector},
    args...,
) = Base.Broadcast.broadcasted(fs, (x...) -> Geometry.UVVector(x...), args...)
 ```
 which produces 16 ambiguities (so it's worse than the current one)

Reply:
I pushed the latest suggestion. However, it still produces 15 ambiguities, as in my previous version